clang/test/CIR/CodeGen/template-specialization.cpp

```diff
@@ -0,0 +1,88 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -Wno-unused-value -fclangir -emit-cir %s -o %t.cir
+// RUN: FileCheck --input-file=%t.cir %s -check-prefix=CIR
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -Wno-unused-value -fclangir -emit-llvm %s -o %t-cir.ll
+// RUN: FileCheck --input-file=%t-cir.ll %s -check-prefix=LLVM
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -Wno-unused-value -emit-llvm %s -o %t.ll
+// RUN: FileCheck --input-file=%t.ll %s -check-prefix=OGCG
+
+template<class T>
+class X {
+public:
+  int f() { return 0; }
+};
+
+template<> class X<int> {
```

Comment:
Add another test for a 'partial' specialization.  There shouldn't be any additional code (since a `ClassTemplatePartialSpecializationDecl` inherits from `ClassTemplateSpecializationDecl`, but it should demonstrative.

You can do this by:

```
template<typename T, typename U>
class Templ {};

template<typename T>
class Templ<T, int>{};

Templ<int, int> t;
```